codegen/src/test/resources/software/amazon/awssdk/codegen/poet/rules/endpoint-provider-class.java

```diff
@@ -57,6 +53,316 @@ private static Map<Identifier, Value> toIdentifierValueMap(QueryEndpointParamete
+                                                            .builder()
+                                                            .fn("isSet")
+                                                            .argv(Arrays.asList(Expr.ref(Identifier
+                                                                                             .of("useFIPSEndpoint"))))
+                                                            .build().validate())
+                                                    .build())
+                                            .addCondition(
+                                                Condition
+                                                    .builder()
+                                                    .fn(FnNode
+                                                            .builder()
+                                                            .fn("booleanEquals")
+                                                            .argv(Arrays.asList(
+                                                                Expr.ref(Identifier
+                                                                             .of("useDualStackEndpoint")),
+                                                                Expr.of(true)))
+                                                            .build().validate())
+                                                    .build())
+                                            .addCondition(
+                                                Condition
+                                                    .builder()
+                                                    .fn(FnNode
+                                                            .builder()
+                                                            .fn("booleanEquals")
+                                                            .argv(Arrays.asList(
+                                                                Expr.ref(Identifier
+                                                                             .of("useFIPSEndpoint")),
+                                                                Expr.of(true)))
+                                                            .build().validate())
+                                                    .build())
+                                            .endpoint(
+                                                software.amazon.awssdk.core.rules.Endpoint
+                                                    .builder()
+                                                    .url(Expr
+                                                             .of("https://query-fips.{region}.{partitionResult#dualStackDnsSuffix}"))
+                                                    .build()),
+                                        Rule.builder()
+                                            .endpoint(
+                                                software.amazon.awssdk.core.rules.Endpoint
```

Comment:
Having two different endpoints on the classpath - software.amazon.awssdk.core.rules.Endpoint and software.amazon.awssdk.core.rules.model.Endpoint - is likely going to be confusing. Should we rename one?